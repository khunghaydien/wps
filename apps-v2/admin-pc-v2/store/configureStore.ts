import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer from '../reducers';

const PRODUCTION = process.env.NODE_ENV === 'production';

const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
        traceLimit: 25,
      })
    : compose;

export default function configureStore(initialState: any) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../reducers', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../reducers').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
