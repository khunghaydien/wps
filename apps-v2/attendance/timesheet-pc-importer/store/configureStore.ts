/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
// 本来は使いたくないが
// Global Container の社員切替が使用しているので
// しかたなく導入している
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.TEST === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middleWares: Middleware[] = [
    // 本来は使いたくないが
    // Global Container の社員切替が使用しているので
    // しかたなく導入している
    thunkMiddleware,
  ];
  if (!PRODUCTION && !TEST) {
    middleWares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middleWares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
