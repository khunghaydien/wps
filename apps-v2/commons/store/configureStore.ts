import { AnyAction, applyMiddleware, createStore, Store } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../reducers';

const createStoreWithMiddleware = applyMiddleware(
  thunkMiddleware,
  logger
)(createStore);

export default function configureStore(
  initialState?: State
): Store<State, AnyAction> {
  const store = createStoreWithMiddleware(rootReducer, initialState);

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      /* eslint-disable  @typescript-eslint/no-var-requires */
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const nextRootReducer = require('../reducers');

      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
}
