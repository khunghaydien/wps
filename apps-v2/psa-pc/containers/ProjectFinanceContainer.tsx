import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

import { JobType } from '@apps/domain/models/psa/BatchJobProject';

import { State } from '@psa/modules';

import {
  clearMemo,
  fetchProjectFinanceInfo,
  getFinanceDetailContractFixed,
  getFinanceDetailContractTnM,
  getFinanceDetailOtherCategory,
  getFinanceDetailOtherExpense,
  getFinanceMemo,
  getWorkDaysFromFinance,
  runBatchJobProject,
  saveFinanceMemo,
} from '@psa/action-dispatchers/Finance';

import ProjectFinancePage from '@psa/components/ProjectScreen/Finance';

const ProjectFinanceContainer = () => {
  const dispatch = useDispatch() as ThunkDispatch<unknown, void, AnyAction>;

  const isFetching = useSelector((state: State) => state.ui.isLoading);
  const currentRoute = useSelector((state: State) => state.ui.siteRoute);
  const projectFinanceOverview = useSelector(
    (state: State) => state.entities.psa.projectFinance.projectFinanceOverview
  );
  const selectedProject = useSelector(
    (state: State) => state.entities.psa.project.project
  );
  const currencyDecimal = useSelector(
    (state: State) => state.userSetting.currencyDecimalPlaces
  );
  const currencySymbol = useSelector(
    (state: State) => state.userSetting.currencySymbol
  );
  const workingDays = useSelector(
    (state: State) => state.entities.psa.projectFinance.workingDays
  );
  const memoValue = useSelector(
    (state: State) => state.entities.psa.projectFinance.note
  );
  const lastModifiedDate = useSelector(
    (state: State) => state.entities.psa.projectFinance.lastModifiedDate
  );
  const getMemo = (noteId: string) => {
    dispatch(getFinanceMemo(noteId));
  };
  const saveMemo = async (
    projectId: string,
    intervalType: string,
    noteId: string,
    note: string,
    summaryInfo?: object,
    detailInfo?: object
  ) => {
    if (note !== '' && memoValue !== note) {
      await dispatch(
        saveFinanceMemo(
          noteId,
          note,
          lastModifiedDate,
          summaryInfo,
          detailInfo,
          true,
          currentRoute
        )
      );
    }
    dispatch(clearMemo());
  };

  const fetchWorkingDays = (intervalType: string) => {
    const { projectId } = selectedProject;
    dispatch(getWorkDaysFromFinance(projectId, intervalType));
  };

  const fetchFinanceDetailContractFixed = (financeCategoryId: string) => {
    dispatch(
      getFinanceDetailContractFixed(
        selectedProject.projectId,
        financeCategoryId
      )
    );
  };

  const fetchFinanceDetailContractTnM = (financeCategoryId: string) => {
    dispatch(
      getFinanceDetailContractTnM(selectedProject.projectId, financeCategoryId)
    );
  };

  const fetchFinanceDetailOtherCategory = (financeCategoryId: string) => {
    const { projectId } = selectedProject;
    dispatch(getFinanceDetailOtherCategory(projectId, financeCategoryId));
  };

  const fetchFinanceDetailOtherExpense = (financeCategoryId: string) => {
    const { projectId } = selectedProject;
    dispatch(getFinanceDetailOtherExpense(projectId, financeCategoryId));
  };

  const selectWeeklyMonthlyView = (intervalType: string) => {
    const { projectId } = selectedProject;
    dispatch(fetchProjectFinanceInfo(projectId, intervalType));
    dispatch(getWorkDaysFromFinance(projectId, intervalType));
  };

  const execBatchJobProject = (jobType: JobType) => {
    const { projectId } = selectedProject;
    dispatch(runBatchJobProject(projectId, jobType));
  };

  return (
    <ProjectFinancePage
      currencyDecimal={currencyDecimal}
      currencySymbol={currencySymbol}
      financeOverviewData={projectFinanceOverview}
      selectedProject={selectedProject}
      fetchFinanceDetailContractFixed={fetchFinanceDetailContractFixed}
      fetchFinanceDetailContractTnM={fetchFinanceDetailContractTnM}
      onSelectWeeklyMonthlyView={selectWeeklyMonthlyView}
      fetchFinanceDetailOtherCategory={fetchFinanceDetailOtherCategory}
      fetchFinanceDetailOtherExpense={fetchFinanceDetailOtherExpense}
      onFetchWorkingDays={fetchWorkingDays}
      workingDays={workingDays}
      getFinanceMemo={getMemo}
      saveFinanceMemo={saveMemo}
      memo={memoValue}
      isFetching={isFetching}
      execBatchJobProject={execBatchJobProject}
    />
  );
};

export default ProjectFinanceContainer;
