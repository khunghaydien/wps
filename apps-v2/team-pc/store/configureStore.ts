/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
