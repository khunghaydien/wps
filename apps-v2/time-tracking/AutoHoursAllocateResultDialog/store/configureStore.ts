/* eslint-disable global-require, import/newline-after-import */
import {
  AnyAction,
  applyMiddleware,
  compose,
  createStore,
  Middleware,
  Store,
} from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(
  initialState?: Partial<State>
): Store<State, AnyAction> {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    middlewares.push(logger);
  }
  const store: Store<State, AnyAction> = createStore(
    rootReducer,
    initialState,
    // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      /* eslint-disable-next-line @typescript-eslint/no-var-requires */
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
