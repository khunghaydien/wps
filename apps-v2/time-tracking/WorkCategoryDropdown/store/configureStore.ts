/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        name: 'WorkCategoryDropdown',
      })
    : compose;

export default function configureStore(initialState?: any) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    middlewares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
