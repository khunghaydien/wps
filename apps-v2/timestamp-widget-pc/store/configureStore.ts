/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    middlewares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
