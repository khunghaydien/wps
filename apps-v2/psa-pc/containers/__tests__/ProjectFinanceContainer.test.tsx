import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { act, cleanup } from '@testing-library/react';

import ProjectFinanceContainer from '../ProjectFinanceContainer';

const mockStore = configureStore([thunk]);

const state = {
  userSetting: {
    companyId: 'test-company',
    currencyDecimalPlaces: 2,
    currencySymbol: '$',
  },
  ui: {
    tab: 'test-tab',
    mode: 'test-mode',
    filter: {},
  },
  entities: {
    psa: {
      setting: {
        useExistingJobCode: true,
      },
      project: {
        pageNum: 1,
        project: {
          projectId: 'test-project',
          projectName: 'test-project-name',
          startDate: '2022-06-28',
          endDate: '2022-06-28',
        },
      },
      role: {
        role: {
          roleTitle: 'test-role',
        },
      },
      resource: {
        resourceList: [],
      },
      projectFinance: {
        workingDays: {
          workdays: [],
          total: 0,
        },
        projectFinanceOverview: {
          revenue: {},
          cost: {},
          margin: {
            intervalTotals: [],
          },
        },
        projectInfo: [
          {
            hidden: true,
            key: 'project-id',
            value: 'a2U9D000000Hdp6UAC',
          },
          {
            key: 'project-title',
            label: 'Project Title',
            value: 'TD QA Project',
          },
          {
            key: 'project-code',
            label: 'Project Code',
            value: 'TDJOB001',
          },
          {
            key: 'project-manager',
            label: 'Project Manager',
            value: 'Sista Aadmi',
          },
          {
            key: 'project-duration',
            label: 'Project Duration',
            value: '2022/04/08 - 2023/04/08',
          },
          {
            key: 'project-client-name',
            label: 'Client Name',
            value: '-',
          },
        ],
        calculations: [
          {
            financeCategory: [
              {
                actual: 0,
                dataKey: 'Cost',
                diff: 0,
                label: 'Labour Cost',
                planned: 0,
              },
              {
                actual: 0,
                dataKey: 'CustomCost',
                diff: 0,
                label: 'Custom Cost',
                planned: 0,
              },
            ],
            financeType: 'Cost',
            total: { actual: 0, diff: 0, label: 'Total', planned: 0 },
          },
          {
            financeCategory: [
              {
                actual: 0,
                dataKey: null,
                diff: 0,
                label: 'Project Revenue',
                planned: 0,
              },
            ],
            financeType: 'Revenue',
            total: { actual: 0, diff: 0, label: 'Total', planned: 0 },
          },
        ],
        grossMargin: {
          actualPercent: 0,
          grossActual: 0,
          grossDifference: 0,
          grossPlanned: 0,
          plannedPercent: 0,
        },
      },
    },
  },
  common: {
    app: {
      loadingDepth: 0,
    },
  },
};

describe('ProjectFinanceContainer Test', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore(state);
    const origDispatch = store.dispatch;
    store.dispatch = jest.fn(origDispatch);
  });

  afterEach(cleanup);

  it('it renders without crashing', async () => {
    await act(() => {
      component = renderer.create(
        <Provider store={store}>
          <ProjectFinanceContainer />
        </Provider>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  // it('dispatch event to fetch finance info', () => {
  //   component = mount(
  //     <Provider store={store}>
  //       <ProjectFinanceContainer />
  //     </Provider>
  //   );
  //   component.find('.ts-psa__finance__refresh-btn').first().simulate('click');
  //   expect(store.dispatch).toHaveBeenCalled();
  // });

  // it('dispatch event to open labour page', () => {
  //   component = mount(
  //     <Provider store={store}>
  //       <ProjectFinanceContainer />
  //     </Provider>
  //   );
  //   const financePage = component.find(ProjectFinancePage).props();
  //   financePage.onClickLabourDetail('Cost');
  //   expect(store.dispatch).toHaveBeenCalled();
  // });

  // it('dispatch event to open finance category page', async () => {
  //   component = mount(
  //     <Provider store={store}>
  //       <ProjectFinanceContainer />
  //     </Provider>
  //   );
  //   const financePage = component.find(ProjectFinancePage).props();
  //   financePage.onClickFinanceDetail('CustomCost', 'cost');
  //   expect(store.dispatch).toHaveBeenCalled();
  // });
});
