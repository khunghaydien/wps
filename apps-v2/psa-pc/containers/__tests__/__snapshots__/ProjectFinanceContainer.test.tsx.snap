// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFinanceContainer Test it renders without crashing 1`] = `
<div
  className="ts-psa__finance "
>
  <textarea
    className="ts-psa__finance__memo"
    onBlur={[Function]}
    onChange={[Function]}
    style={
      Object {
        "left": "0px",
        "maxWidth": "200px",
        "minHeight": "100px",
        "position": "absolute",
        "top": "0px",
        "visibility": "hidden",
        "zIndex": "999",
      }
    }
  />
  <div
    className="ts-psa__common-header"
  >
    <div
      className="ts-psa__common-header-left"
    >
      <span
        className="ts-psa__common-header-breadcrumb"
      />
      <span
        className="ts-psa__common-header-title"
      >
        Financials
      </span>
    </div>
    <div
      className="ts-psa__common-header__btn-area"
    >
      <button
        className="ts-psa__finance__execute-btn ts-button ts-button--secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
        value=""
      >
        <div
          className="ts-button__contents"
        >
          <span
            className="ts-button__text"
          >
            Calculate
             
            <br />
             
            <span
              className="ts-psa__finance__execute-time"
            >
              Last Time
              : 
            </span>
          </span>
        </div>
      </button>
      <button
        className="ts-psa__finance__download-btn ts-button ts-button--default"
        data-testid="ts-psa__finance__download-btn"
        disabled={false}
        onClick={[Function]}
        type="button"
        value=""
      >
        <div
          className="ts-button__contents"
        >
          <span
            className="ts-button__text"
          >
            <test-file-stub
              className="ts-psa__finance__download-icon"
            />
          </span>
        </div>
      </button>
      <div
        className="ts-psa__toggle-view"
      >
        <button
          className="ts-psa__toggle-view__refresh-btn ts-button ts-button--default"
          data-testid="ts-psa__toggle-view__refresh-btn"
          disabled={false}
          onClick={[Function]}
          type="button"
          value=""
        >
          <div
            className="ts-button__contents"
          >
            <span
              className="ts-button__text"
            >
              <test-file-stub />
            </span>
          </div>
        </button>
        <div
          className="ts-psa__toggle-view__select-view-mode"
        >
          <button
            className="ts-button ts-button--default"
            data-testid="ts-psa__toggle-view__btn--week"
            disabled={false}
            onClick={[Function]}
            type="button"
            value=""
          >
            <div
              className="ts-button__contents"
            >
              <span
                className="ts-button__text"
              >
                Week
              </span>
            </div>
          </button>
          <button
            className="is-active ts-button ts-button--default"
            data-testid="ts-psa__toggle-view__btn--month"
            disabled={false}
            onClick={[Function]}
            type="button"
            value=""
          >
            <div
              className="ts-button__contents"
            >
              <span
                className="ts-button__text"
              >
                Month
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ts-psa__project-information-header__header-info js-is-expanded"
  >
    <button
      className="ts-psa__project-information-header__header-info--buttons"
      onClick={[Function]}
    >
      <span
        className="ts-psa__project-information-header__header-info__toggle-icon-bg"
      >
        <test-file-stub
          className="ts-psa__project-information-header__header-info__toggle-icon"
        />
      </span>
      <span>
        Close Project Information
      </span>
    </button>
    <div
      className="ts-psa__project-information-header__header-info__column"
    >
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            {$Psa_Clbl_ProjectCode}
            :
          </label>
        </span>
      </div>
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            Project Title
            :
          </label>
        </span>
      </div>
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            Project Duration
            :
          </label>
          06/28/2022 - 06/28/2022
        </span>
      </div>
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            Contract Amount
            :
          </label>
           0.00
        </span>
      </div>
    </div>
    <div
      className="ts-psa__project-information-header__header-info__column"
    >
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            Project Manager
            :
          </label>
        </span>
      </div>
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            Client
            :
          </label>
          -
        </span>
      </div>
      <div>
        <span
          className="ts-psa__project-information-header__header-info-item"
        >
          <label
            className="ts-psa__project-information-header__header-info-label"
          >
            Status
            :
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ts-psa__finance__revenue-overview"
  >
    <div
      className="ts-psa__finance__revenue-overview__column"
    >
      <div
        className="ts-psa__finance__revenue-overview__column-item"
      >
        <div
          className="ts-psa__finance__revenue-amt"
        >
          <div
            className="ts-psa__finance__revenue-amt-label"
          >
            Actual Revenue
          </div>
          <div
            className="ts-psa__finance__revenue-amt-amount"
            title={0}
          >
            0
          </div>
        </div>
      </div>
      <div
        className="ts-psa__finance__revenue-overview__column-item"
      >
        <div
          className="ts-psa__finance__revenue-amt"
        >
          <div
            className="ts-psa__finance__revenue-amt-label"
          >
            Target Revenue
          </div>
          <div
            className="ts-psa__finance__revenue-amt-amount"
            title={0}
          >
            0
          </div>
        </div>
      </div>
      <div
        className="ts-psa__finance__revenue-overview__column-item"
      >
        <div
          className="ts-psa__finance__revenue-amt"
        >
          <div
            className="ts-psa__finance__revenue-amt-label"
          >
            Target Achievement
          </div>
          <div
            className="ts-psa__finance__revenue-amt-amount"
            title="0%"
          >
            0%
          </div>
        </div>
      </div>
    </div>
    <div
      className="ts-psa__finance__revenue-overview__column"
    >
      <div
        className="ts-psa__finance__revenue-overview__column-item"
      >
        <div
          className="ts-psa__finance__revenue-amt"
        >
          <div
            className="ts-psa__finance__revenue-amt-label"
          >
            Actual Margin
          </div>
          <div
            className="ts-psa__finance__revenue-amt-amount"
            title="0%"
          >
            0%
          </div>
        </div>
      </div>
      <div
        className="ts-psa__finance__revenue-overview__column-item"
      >
        <div
          className="ts-psa__finance__revenue-amt"
        >
          <div
            className="ts-psa__finance__revenue-amt-label"
          >
            Target Margin
          </div>
          <div
            className="ts-psa__finance__revenue-amt-amount"
            title="0%"
          >
            0%
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ts-psa__empty-screen"
  >
    <div
      className="ts-psa__empty-screen__header"
    >
      <span>
        Looks like there is no data here.
      </span>
    </div>
    <div
      className="ts-psa__empty-screen__body"
    />
  </div>
</div>
`;
